xchange-core/src/main/java/org/knowm/xchange/derivative/OptionsContract.java

```diff
@@ -0,0 +1,132 @@
+package org.knowm.xchange.derivative;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+import org.knowm.xchange.currency.CurrencyPair;
+import org.knowm.xchange.instrument.Instrument;
+
+import java.io.Serializable;
+import java.math.BigDecimal;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.Objects;
+
+public class OptionsContract extends Instrument 
+      implements Derivative, Comparable<OptionsContract>, Serializable {
+
+  private static final long serialVersionUID = 1L;    
+    
+  public enum OptionType {
+    CALL("C"),
+    PUT("P");
+
+    private final String postfix;
+
+    OptionType(String postfix) {
+      this.postfix = postfix;
+    }
+    
+    public static OptionType fromString(String s) {
+      if(CALL.postfix.equalsIgnoreCase(s)) return CALL;
+      if(PUT.postfix.equalsIgnoreCase(s)) return PUT;
+      throw new IllegalArgumentException("Unknown option type: " + s);
+    }
+    
+  }
+
+  private static final ThreadLocal<DateFormat> DATE_PARSER =
+      ThreadLocal.withInitial(() -> new SimpleDateFormat("yyMMdd"));
+
+  private static final Comparator<OptionsContract> COMPARATOR =
+      Comparator.comparing(OptionsContract::getCurrencyPair)
+        .thenComparing(OptionsContract::getExpireDate)
```

Comment:
Is it possible to make this consistent with the futures contact and call the expireDate  prompt, as it is not always the exact expiry date but more a symbol representing an exact date i.e. `sept` really mean the 3rd friday in sept.

Reply:
It is consistent with current FuturesContract implementation, where java.util.Date is used as well